DoSomething/rogue
# Adds /reviews endpoint to update photo(s)'s status when it gets reviewed and updates signups/photos

#### What's this PR do?
- Adds update functionality to update a signup (`why_participated` and `quantity`)
- Adds update functionality to add a new photo post to a signup 
- Adds `/reviews` endpoint to update a photo(s)'s status when reviewed by an admin

#### How should this be reviewed?
This should be tested with [corresponding Phoenix work](https://github.com/DoSomething/phoenix/pull/7295).

1.Change Rogue API version number field to v2 in the Phoenix Rogue API settings.
2.  Run `drush ubdb` in drupal to reflect new database changes.

Make sure all information matches up between Phoenix and Rogue databases for below scenarios:
- Create a new signup (RB) and photo (reportback item) from Phoenix web.
- Add another photo (rb item).
- Add another photo (rb item) and update the signup (reportback's quantity and why_participated).
- Only update the signup (reportback's quantity and why_participated).
- Review a photo (reportback item's status as an admin).
- Cron job for reportbacks and reportback items.

#### Any background context you want to provide?
A few notes: 
1. If we're adding a new photo AND changing the signup's `quantity` and/or `why_participated`, this is all one event because we don't have a good way differentiate  if it is only adding a new photo or adding new photo and changing `quantity` and/or `why_participated` at the moment e.g. is it sufficient to use character count to see if `why_participated` has changed? This will be for decisions to make in the future cc @ngjo @jessleenyc 
2. In this PR, the signup's `event_id` that gets returned is the original `event_id` when the signup was created, not the `event_id` of when the signup was updated. A new event is still created when a signup update occurs. However, the `event_type` is a `post_photo` since we mentioned not wanting to complicate this now. We don't have a great way of tying this signup update back to the signup. @sbsmith86 @katiecrane Do you think this is ok for now until we make more decisions? 
3. We don't have a `reviewer` column in the `photos` table like we do in the `reportback_items` table and don't have information on who reviewed it. However, when a photo is reviewed, I've changed the `submission_type` of the event to be `admin review`. I can add: 
  - add more info here instead of the hard coded `admin review` (admin's name/email/northstar id/etc) in this column 
  - add `reviewer` to the `photo` table (mimicking the `reportback_items` table)?
  - change `event_type` to `admin review` (wasn't sure about this since we talked about keeping `event_type`s as only `post_photo` for now and make decisions about this later cc @sbsmith86 @katiecrane 
4. When an admin reviews a photo, right now we have no idea which photo it is. It only creates a new event and the only info we have is that it is a `submission_type` `admin_review`. Do we want to add something to the `events` table to refer to which post is it? Then it is a little tricky though because we'd have to add `photo_id` or `video_id` etc? cc @sbsmith86 @katiecrane 

#### Relevant tickets
Fixes #100 

#### Checklist
